erpnext/public/js/bank_reconciliation_tool/dialog_manager.js

```diff
@@ -35,13 +35,590 @@ erpnext.accounts.bank_reconciliation.DialogManager = class DialogManager {
+				args: {
+					bank_transaction_name: this.bank_transaction.name,
+					reference_number: values.reference_number,
+					reference_date: values.reference_date,
+					party_type: values.party_type,
+					party: values.party,
+					posting_date: values.posting_date,
+					mode_of_payment: values.mode_of_payment,
+					entry_type: values.journal_entry_type,
+					second_account: values.second_account,
+				},
+				callback: (response) => {
+					const alert_string = __("Bank Transaction {0} added as Journal Entry", [this.bank_transaction.name]);
+					frappe.show_alert(alert_string);
+					this.update_dt_cards(response.message);
+					this.dialog.hide();
+				},
+			});
+		}
+	
+		update_transaction(values) {
+			frappe.call({
+				method:
+					"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.update_bank_transaction",
+				args: {
+					bank_transaction_name: this.bank_transaction.name,
+					reference_number: values.reference_number,
+					party_type: values.party_type,
+					party: values.party,
+				},
+				callback: (response) => {
+					const alert_string = __("Bank Transaction {0} updated", [this.bank_transaction.name]);
+					frappe.show_alert(alert_string);
+					this.update_dt_cards(response.message);
+					this.dialog.hide();
+				},
+			});
+		}
+	
+		edit_in_full_page() {
+			const values = this.dialog.get_values(true);
+			if (values.document_type == "Payment Entry") {
+				frappe.call({
+					method:
+						"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_payment_entry_bts",
+					args: {
+						bank_transaction_name: this.bank_transaction.name,
+						reference_number: values.reference_number,
+						reference_date: values.reference_date,
+						party_type: values.party_type,
+						party: values.party,
+						posting_date: values.posting_date,
+						mode_of_payment: values.mode_of_payment,
+						project: values.project,
+						cost_center: values.cost_center,
+						allow_edit: true
+					},
+					callback: (r) => {
+						const doc = frappe.model.sync(r.message);
+						frappe.set_route("Form", doc[0].doctype, doc[0].name);
+					},
+				});
+			} else {
+				frappe.call({
+					method:
+						"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_journal_entry_bts",
+					args: {
+						bank_transaction_name: this.bank_transaction.name,
+						reference_number: values.reference_number,
+						reference_date: values.reference_date,
+						party_type: values.party_type,
+						party: values.party,
+						posting_date: values.posting_date,
+						mode_of_payment: values.mode_of_payment,
+						entry_type: values.journal_entry_type,
+						second_account: values.second_account,
+						allow_edit: true
+					},
+					callback: (r) => {
+						var doc = frappe.model.sync(r.message);
+						frappe.set_route("Form", doc[0].doctype, doc[0].name);
+					},
+				});
+			}
+		}
+	
+	};
```

Comment:
What is this for, can't see any mention in the description. Please be more descriptive